restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +95,96 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  private static Object fillInDataDefault(DataSchema schema, Object dataWithoutDefault)
+  {
+    switch (schema.getType())
+    {
+      case RECORD:
+        return fillInDefaultOnRecord((RecordDataSchema) schema, (DataMap) dataWithoutDefault);
+      case TYPEREF:
+        return fillInDefaultOnTyperef((TyperefDataSchema) schema, dataWithoutDefault);
+      case MAP:
+        return fillInDefaultOnMap((MapDataSchema) schema, (DataMap) dataWithoutDefault);
+      case UNION:
+        return fillInDefaultOnUnion((UnionDataSchema) schema, (DataMap) dataWithoutDefault);
+      case ARRAY:
+        return fillInDefaultOnArray((ArrayDataSchema) schema, (DataList) dataWithoutDefault);
+      default:
+        return dataWithoutDefault;
+    }
+  }
+
+  public static DataMap fillInDefaultOnRecord(RecordDataSchema schema, DataMap dataMap)
+  {
+    DataMap dataWithDefault = new DataMap(DataMapBuilder.getOptimumHashMapCapacityFromSize(dataMap.size()));
+    dataWithDefault.putAll(dataMap);
```

Comment:
We should use CheckedUtil::*withoutChecking methods.

By default DataMap and DataList check for cycles when a new object is added. However, since we are controling the logic and are not creating cycles, we can skip those checks for performance.

Reply:
Sure